activation/atxdb.go

```diff
@@ -45,6 +46,9 @@ func (db *ActivationDb) ProcessBlockATXs(blk *types.Block) {
 // ProcessAtx validates the active set size declared in the atx, and validates the atx according to atx validation rules
 // it then stores the atx with flag set to validity of the atx
 func (db *ActivationDb) ProcessAtx(atx *types.ActivationTx) {
+	db.processAtxMutex.Lock()
+	defer db.processAtxMutex.Unlock()
```

Comment:
Just reminding you again - defer is very slow and should be avoided. Consider taking care of all returns instead.